src/Decoder.ts

```diff
@@ -12,10 +12,18 @@ const enum State {
   MAP_VALUE,
 }
 
+type MapKeyType = string | number;
+
+const isValidMapKeyType = (key: unknown): key is MapKeyType => {
```

Comment:
Maybe we don't need this function and I can just put it into if. It just looks better :)

This function should not affect performance because it's very small and will be inlined